src/main/java/uk/nhs/ctp/service/CarePlanService.java

```diff
@@ -2,25 +2,27 @@
 
 import java.util.List;
 import java.util.stream.Collectors;
-import lombok.AllArgsConstructor;
-import lombok.extern.slf4j.Slf4j;
+import java.util.stream.Stream;
+import lombok.RequiredArgsConstructor;
 import org.hl7.fhir.dstu3.model.Bundle;
 import org.hl7.fhir.dstu3.model.Bundle.BundleEntryComponent;
 import org.hl7.fhir.dstu3.model.CarePlan;
+import org.hl7.fhir.dstu3.model.CarePlan.CarePlanStatus;
 import org.hl7.fhir.dstu3.model.ReferralRequest;
 import org.hl7.fhir.dstu3.model.ResourceType;
 import org.springframework.stereotype.Service;
+import uk.nhs.ctp.service.fhir.AuthenticatedStorageService;
 import uk.nhs.ctp.service.fhir.ReferenceService;
 import uk.nhs.ctp.service.fhir.StorageService;
 import uk.nhs.ctp.utils.RetryUtils;
 
 @Service
-@AllArgsConstructor
-@Slf4j
+@RequiredArgsConstructor
 public class CarePlanService {
 
-  private StorageService storageService;
-  private ReferenceService referenceService;
+  private final StorageService storageService;
+  private final ReferenceService referenceService;
+  private final AuthenticatedStorageService authenticatedStorageService;
```

Comment:
doesn't the EMS already have the interceptors to add the token on its clients by default in FHIRConfig? If we're replacing that at least needs a tech debt?

Reply:
Agreed, there were two versions of this: one using the AuthRestInterceptor (EMS) and one using the AuthenticatedFhirClientFactory (other services). This change brings the EMS in line with the other services using the common package, but unless you're planning to replace it everywhere it's inconsistent with how the EMS fetches resources elsewhere. Also, I feel like the interceptor is a neater solution since it doesn't require all the code to care about the FhirClientFactory stuff - if we were to change to one solution or the other I think I prefer the interceptor.

Reply:
Ok, agreed. I wasn't aware that FHIRConfig did this, but I do think there's value in also physically reflecting the theoretical distinction between the EMS and its auth server role. I've raised [CDSCT-424](https://nhsd-jira.digital.nhs.uk/browse/CDSCT-424) to cover this.

Reply:
Yep, that part could definitely still be applied in the `SupplierTokenResolver`

Reply:
This still doesn't cover my original question - if we're going to refactor to use this - we need a tech debt to use the authenticated storage service everywhere and remove the old one.

Reply:
I think for now we should just use the existing StorageService, since AuthenticatedStorageService just adds a second interceptor on top of the default behaviour. Work to separate the logic should be left to the new TD ticket.

Reply:
Ok, makes sense - I'll remove the `AuthenticatedStorageService`.